components/wallet-logger.js

```diff
@@ -313,9 +315,17 @@ export function useWalletLogs (wallet, initialPage = 1, logsPerPage = 10) {
     }
   }, [logs, wallet?.def, loadLogsPage])
 
-  useInterval(() => {
-    loadNew().catch(console.error)
-  }, 1_000, [loadNew])
+  useEffect(() => {
+    // only fetch new logs if we are on a page that uses logs
+    const needLogs = router.asPath.startsWith('/settings/wallets') || router.asPath.startsWith('/wallet/logs')
+    if (me && needLogs) {
+      // this will poll the logs _every_ second instead of just once since a poll
+      // changes `loadNew` so we will run the effect again with a new timeout after each poll
```

Comment:
Is a side effect of a side effect the best way to do this?

Reply:
When you say it like that, probably not. But afaict, if we want to keep `loadNew`, we have only these two options:

- add code that only looks like it gets run but doesn't actually get run (how it was before with `useInterval` or a `setTimeout` callback that calls itself again)
- adding a comment that mentions this side effect of a side effect without that code (what I did here)

From these options, I think a comment is better.

If you prefer to have this side effect removed completely (not use something like `loadNew`), we would need to find a way to pass data from the previous poll (= previous effect) to the next poll (= next effect) _without changing the effect_. But since the next poll should always use the latest available timestamp in the logs ... I am not sure how we can use `useState` for logs but still have different state in our polls in `useEffect`.

And I don't think we want to use `useRef` for this.

This is at least my understanding so far.

Reply:
> mhh, we could use useRef only for the latest timestamp. We don't need to replace useState for the logs with useRef for the next poll to know what the latest timestamp since that's all it needs 🤔

did that in f865488